src/System.Private.CoreLib/src/System/Threading/Timer.cs

```diff
@@ -469,24 +540,19 @@ internal bool Change(uint dueTime, uint period)
                 if (m_canceled)
                     throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
 
-                // prevent ThreadAbort while updating state
-                try { }
-                finally
-                {
-                    m_period = period;
+                m_period = period;
 
-                    if (dueTime == Timeout.UnsignedInfinite)
-                    {
-                        m_associatedTimerQueue.DeleteTimer(this);
-                        success = true;
-                    }
-                    else
-                    {
-                        if (FrameworkEventSource.IsInitialized && FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
-                            FrameworkEventSource.Log.ThreadTransferSendObj(this, 1, string.Empty, true);
+                if (dueTime == Timeout.UnsignedInfinite)
+                {
+                    m_associatedTimerQueue.DeleteTimer(this);
+                    success = true;
+                }
+                else
+                {
+                    if (FrameworkEventSource.IsInitialized && FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
+                        FrameworkEventSource.Log.ThreadTransferSendObj(this, 1, string.Empty, true);
```

Comment:
We should add an additional optional int parameters to the ThreadTransferSendObj at the end and then pass the duration when logging this event.    This would give us useful information from the field about the statistics of durations.

Reply:
> We should add an additional optional int parameters to the ThreadTransferSendObj at the end and then pass the duration when logging this event.

Sounds good, but let's do it separately from this PR.

As for the mechanics, it's ok to add arguments to an existing event like that?  How does one define an optional argument with EventSource?

Reply:
I am OK with a separate PR.   You can add new arguments to the END of an event (because any existing parsers simply ignore the 'rest' of the payload).   

I have never actually used an optional argument for an eventSource method, but it should work as the IL will have the extra arg, the the compiler just knows to add it at the call site as needed.  However in this case the issue is mute because there is already a 'wrapper' method around the real logging method because we need to convert the Timer Object to a ID.   Thus you can make it optional in the wrapper, and required in the logging method.

Reply:
> Thus you can make it optional in the wrapper, and required in the logging method.

Sorry, I know how to make something optional in C# / the IL.  I thought you were suggesting of making it optional in the actual event, i.e. that there was some kind of optional notion in ETW and that could be represented in an [Event(...)] description.  If it's ok to just update the [Event(...)] with an additional argument and that won't break a consumer, then no problem.